File: jmolecules-axon/02-onion-cqrs-es-bank/src/main/java/org/jmolecules/example/axonframework/domain/command/model/MoneyTransferIdGenerator.java
package org.jmolecules.example.axonframework.domain.command.model;

import java.util.UUID;
import java.util.function.Supplier;

public interface MoneyTransferIdGenerator extends Supplier<String> {

  MoneyTransferIdGenerator RANDOM = () -> UUID.randomUUID().toString();

}

File: jmolecules-axon/03-onion-axon-cqrs-es-bank/src/main/kotlin/org/jmolecules/example/axonframework/core/usecase/moneytransfer/MoneyTransferIdGenerator.kt
package org.jmolecules.example.axonframework.core.usecase.moneytransfer

import org.jmolecules.example.axonframework.core.model.moneytransfer.MoneyTransferId
import java.util.function.Supplier

interface MoneyTransferIdGenerator : Supplier<MoneyTransferId>
